core/src/main/java/io/fabric8/maven/core/config/ResourceConfig.java

```diff
@@ -154,6 +154,29 @@ public ConfigMap getConfigMap() {
     public static class Builder {
         private ResourceConfig config = new ResourceConfig();
 
+        public Builder() { }
+
+        public Builder(ResourceConfig config) {
+            if(config != null) {
+                this.config.env = config.getEnv().orElse(null);
+                this.config.controllerName = config.getControllerName();
```

Comment:
Thanks, I think this is good enough (although its not really a complete deep-copy, as you would have to create new collections, too)